### PR TITLE
Add telemetry on/off status to turnstile event

### PR DESCRIPTION
**Pull request by friedbunny**
Depends on #3495.

/cc @1ec5

### CODE REVIEW

File: platform/ios/src/MGLMapboxEvents.m [hunk 1]
```diff
@@ -391,31 +391,27 @@ - (void)startUpdatingLocation {
...
-            [strongSelf flush];
+        NSString *idfv = [[[UIDevice currentDevice] identifierForVendor] UUIDString];
+        NSDictionary *vevt = @{
+            @"event" : MGLEventTypeAppUserTurnstile,
+            @"created" : [strongSelf.rfc3339DateFormatter stringFromDate:[NSDate date]],
+            @"appBundleId" : strongSelf.appBundleId,
+            @"vendorId": idfv,
+            @"version": @(version),
+            @"instance": strongSelf.instanceID,
+            @"telemetry": @([[strongSelf class] isEnabled])
```
Comment (1ec5) [inline platform/ios/src/MGLMapboxEvents.m]: Perhaps a more descriptive key like `telemetryEnabled` would be more future-proof.

**Comment by mikemorris** [comment]
:+1: Will be good to surface these stats.

File: platform/ios/src/MGLMapboxEvents.m (hunk 1 above)
Comment (friedbunny) [inline platform/ios/src/MGLMapboxEvents.m]: `enabled.telemetry` would fit with similar key names. 🤔

**Comment by friedbunny** [comment]
Superceded by #3620.

**Pull request closed without merging**